Properly detect remote mounted file systems with /usr/gnu/bin/df on Solaris.
See the comments in CR# 15595871 for more details.

This patch has not been passed upstream yet. It needs to be reworked into
a format that would be acceptable.

--- lib/mountlist.c
+++ lib/mountlist.c
@@ -231,13 +231,13 @@ me_remote (char const *fs_name, _GL_UNUS
 #endif
 
 #ifndef ME_REMOTE
-/* A file system is "remote" if its Fs_name contains a ':'
-   or if (it is of type (smbfs or cifs) and its Fs_name starts with '//')
-   or if it is of any other of the listed types
+/* A file system is "remote" if its Fs_name contains a ':' (and isn't of
+   type pcfs) or if (it is of type (smbfs or cifs) and its Fs_name starts
+   with '//') or if it is of any other of the listed types
    or Fs_name is equal to "-hosts" (used by autofs to mount remote fs).
    "VM" file systems like prl_fs or vboxsf are not considered remote here. */
 # define ME_REMOTE(Fs_name, Fs_type)            \
-    (strchr (Fs_name, ':') != NULL              \
+    ((strchr (Fs_name, ':') != NULL && strcmp (Fs_type, "pcfs") != 0) \
      || ((Fs_name)[0] == '/'                    \
          && (Fs_name)[1] == '/'                 \
          && (strcmp (Fs_type, "smbfs") == 0     \
